torch/csrc/jit/frontend/script_type_parser.cpp

```diff
@@ -26,8 +26,15 @@ TypePtr ScriptTypeParser::subscriptToType(
     const std::string& typeName,
     const Subscript& subscript) const {
   if (typeName == "Tuple") {
+    auto subscript_exprs = subscript.subscript_exprs();
+    // Arbitrary-length homogeneous tuple as defined in PEP-484
+    if (subscript_exprs.size() == 2 && subscript_exprs[1].kind() == TK_DOTS) {
+      auto elem_type = parseTypeFromExprImpl(subscript_exprs[0]);
+      // Alias arbitrary-length tuple to ListType
+      return ListType::create(elem_type);
```

Comment:
I think treating an any-length homogenous tuple as a list isn't quite correct. List types are mutable, while tuples are immutable. Consider the following code:
```
def foo(a: Tuple[int, ...]) -> int:
    return a[0] = 1
```
In the proposed change, TorchScript would allow this (since `a` would just be a `List[int]` to the compiler), but Python would not, since item assignment is disallowed on tuples.

If we want to support this type annotation correctly, we probably need to come up with a new type with the correct behavior (basically, an immutable sequence of statically unknown length).

Reply:
Good point, although not a very good example, as torch.script now misses those, which should be fixed by:
https://github.com/pytorch/pytorch/pull/44929